diff --git a/node_modules/react-native-create-thumbnail/android/src/main/java/com/createthumbnail/CreateThumbnailModule.java b/node_modules/react-native-create-thumbnail/android/src/main/java/com/createthumbnail/CreateThumbnailModule.java
index 9bd67d8..e8cada2 100644
--- a/node_modules/react-native-create-thumbnail/android/src/main/java/com/createthumbnail/CreateThumbnailModule.java
+++ b/node_modules/react-native-create-thumbnail/android/src/main/java/com/createthumbnail/CreateThumbnailModule.java
@@ -189,8 +189,12 @@ public class CreateThumbnailModule extends ReactContextBaseJavaModule {
         }
 
         Bitmap image = retriever.getFrameAtTime(time * 1000, MediaMetadataRetriever.OPTION_CLOSEST_SYNC);
+      try {
         retriever.release();
-        if (image == null) {
+      } catch (IOException e) {
+        throw new RuntimeException(e);
+      }
+      if (image == null) {
             throw new IllegalStateException("File doesn't exist or not supported");
         }
         return image;
